test/functional/test-video-manager.js

```diff
@@ -46,24 +48,139 @@ describes.fakeWin('VideoManager', {
 }, env => {
   let sandbox;
   let videoManager;
+  let klass;
+  let video;
+  let impl;
+  let spy;
 
   it('should register common actions', () => {
-    const klass = createFakeVideoPlayerClass(env.win);
-    const video = env.createAmpElement('amp-test-fake-videoplayer', klass);
-    const impl = video.implementation_;
-    const spy = sandbox.spy(impl, 'registerAction');
-    videoManager.register(impl);
-
     expect(spy).to.have.been.calledWith('play');
     expect(spy).to.have.been.calledWith('pause');
     expect(spy).to.have.been.calledWith('mute');
     expect(spy).to.have.been.calledWith('unmute');
   });
 
+  it('should be paused if autoplay is not set', () => {
+    impl.hasAutoplay_ = false;
+    impl.isVisible_ = false;
+    const curState = videoManager.getPlayingState(impl);
+    expect(curState).to.equal(PlayingStates.PAUSED);
+  });
+
+
+  it('autoplay - should be manually playing if user interacted', () => {
+    impl.hasAutoplay_ = true;
+    impl.isVisible_ = false;
+    impl.play();
+    listenOncePromise(video, VideoEvents.PLAY).then(() => {
+      const curState = videoManager.getPlayingState(impl);
+      expect(curState).to.equal(PlayingStates.PLAYING_MANUAL);
+    });
+  });
+
+  it('autoplay - should be paused if the user pressed pause after playing',
+  () => {
+    impl.hasAutoplay_ = true;
+    impl.isVisible_ = false;
+    impl.play();
+    listenOncePromise(video, VideoEvents.PLAY).then(() => {
+      impl.pause();
+      listenOncePromise(video, VideoEvents.PAUSE).then(() => {
+        const curState = videoManager.getPlayingState(impl);
+        expect(curState).to.equal(PlayingStates.PAUSED);
+      });
+    });
+  });
+
+  it('autoplay - initially there should be no user interaction', () => {
+    impl.hasAutoplay_ = true;
+    impl.isVisible_ = false;
+    const userInteracted = videoManager.userInteractedWithAutoPlay(impl);
+    expect(userInteracted).to.be.false;
+  });
+
+
+  it('autoplay - should register user interaction', () => {
+    impl.hasAutoplay_ = true;
+    impl.isVisible_ = false;
+    impl.play();
```

Comment:
this  won't pass after adding `return`, calling `play` on video player is not considered a user interaction. Only a `click` on the `mask` overlay is.